src/fstext/fstext-utils-inl.h

```diff
@@ -524,27 +524,22 @@ bool FollowingInputSymbolsAreSameClass(bool end_is_epsilon, const Fst<Arc> &fst,
   return true;
 }
 
+// TODO(galv): Confirm that start_is_epsilon is no longer necessary
```

Comment:
Hm, I think start_is_epsilon may still be necessary.   The issue is that if you had a topology which as a phone acceptor was like:
```
0  0  ah 
1  0.0
```
then the start state would mean both "no preceding phone", and "we just finished phone ah".  Imagine ah has just one state, and you'll see that the start state would get a self-loop on it, which isn't right if this is the first phone.

Reply:
I am still having trouble with this. We ban self loops on the start state in our phone topologies. Therefore, I don't think that your case would ever happen.

I am guessing that you really meant this:

```
0  0  ah 
1  0.0
```

Reply:
No arcs are allowed with a destination of the start state, either. I think we are fine to remove this.

Reply:
The phone topology itself is not allowed to have self-loops at the start state, but the issue is that in the larger graph topology, there may be arcs entering the start state; and if there is exactly one such arc, it would lead to a self-loop being added to the start state (reflecting the transition-id on the entering arc).

Reply:
(I'm surprised that github hasn't posted your message here @danpovey). I understand now,  and I will make sure to keep the start_is_epsilon parameter (along with the code it is associated with) in the function.

Reply:
Thanks!!
Obviously, to get the example scripts working, some changes will be needed, especially as involves transition-scale/self-loop-scale (removing them); removing code about training or applying the learned transition-probs; modifying the prototype topologies to correspond with the new format.